ide/projectui/src/org/netbeans/modules/project/ui/ProjectUtilities.java

```diff
@@ -598,12 +596,12 @@ public static Set<FileObject> openProjectFiles (Project p, Group grp) {
         return toRet;
     }
     
-    private static Set<String> getOpenFilesUrls(Project p, String groupName) {
+    private static List<String> getOpenFilesUrls(Project p, String groupName) {
```

Comment:
Is there any reason to switch the types from Set to List (here and other places)? If for some reason the old code relied on Set removing duplicate entries, then those duplicates will now start appearing.

(Ideally there'd be an "OrderedSet" type, but there is not.)

In general I'm a fan of always just using LinkedHashSet instead of HashSet, except if there are huge performance benefits to HashSet (very rare, in fact it's faster to iterate over a LinkedHashSet than a HashSet). Since HashSet gives an undefined order, it is always safe to replace HashSet with LinkedHashSet. But switching Set to List can change behavior.

Reply:
reason was iteration convenience since two collections had to be compared index by index. I do agree that List vs Set is a statement in itself.

I did also wonder which methods were meant to reduplicate and which were not and planned to give this another pass while checking for that aspect.

This might stay a list (due to convenience) - but a reduplicated one. (will also add a comment).

Reply:
Right, i did think about this while making the changes - but I didn't know the spec and there were no comments in code.

So what I likely will end up doing is to send a deduplicated list back + add a comment. This has probably the lowest risk of not breaking anything since the old behavior would be similar.

Reply:
@neilcsmith-net Oh that's true. I think either retaining the duplication or keeping only the first occurrence would be fine. But would be a good case to test, just to see that nothing breaks.

> This might stay a list (due to convenience) - but a reduplicated one. (will also add a comment).

Yeah, sure. (I assume you meant "de"duplicated.)

Reply:
btw how can I open the same file twice

Reply:
Tab menu, "clone" IIRC.